target/scripts/check-for-changes.sh

```diff
@@ -43,10 +45,13 @@ while true
 do
   LOG_DATE=$(date +"%Y-%m-%d %H:%M:%S ")
 
+  # Lock configuration while working
+  create_lock "${SCRIPT_NAME}"
+
   # get chksum and check it, no need to lock config yet
   _monitored_files_checksums >"${CHKSUM_FILE}.new"
-
-  if ! cmp --silent -- "${CHKSUM_FILE}" "${CHKSUM_FILE}.new"
+  cmp --silent -- "${CHKSUM_FILE}" "${CHKSUM_FILE}.new"
+  if [ $? -eq 1 ]
```

Comment:
Is there a benefit in doing it this way? I did a quick grep on the project (excluded the tests) and could find only one other instance of a `$?` check. So to keep it uniform across the project, I would leave it as it was originally.

Reply:
I thought about this line too. But I think there was this „issue“ where `if ! cmp …` will always be true when `cmp` does not report a status code of 0. This way, the check specifically checks whether the status code is 1. @NorseGaud is that about right?

Reply:
Correct, `cmp` returns specific exit codes for certain behavior/reasons. We need to be checking that it is truly a failure for the right reason.

Reply:
Ack. For reference:

Return values

    0 – files are identical
    1 – files differ
    2 – inaccessible or missing argument

Reply:
We should add this as comments above. Also, I'd like to see `bash`'s `[[ ... ]]` instead of `sh`'s `[ ... ]` :)

Reply:
@georglauterbach I'd rather keep the posix compliance flexibility of [] if possible. I've had different behavior between shells when using [[ ]] to test integer equals in the past and it gives me a lot of anxiety 😅  Is it ok if we leave it how it is?

Reply:
Hmmm. I guess we could leave it at this point as is, but isn’t this code executed _inside_ the container and therefore with a well-behaving shell (Bash 5)?

Rationale: I‘d like to stay to with the style guidelines which are not POSIX compliant. The container does a lot of numerical comparisons in other places too which have not troubled anyone :)

Reply:
But we can leave this as is for now. I don't want to dismiss the reviews again. :D